emission/net/ext_service/habitica/proxy.py

```diff
@@ -153,10 +153,10 @@ def setup_party(user_id, group_id_from_url, inviterId):
   return group_id
 
 def setup_default_habits(user_id):
-  bike_habit = {'type': "habit", 'text': "Bike", 'notes': "3 km = 1+ (calculated automatically)", 'up': True, 'down': False, 'priority': 2}
-  bike_habit_id = create_habit(user_id, bike_habit)
-  walk_habit = {'type': "habit", 'text': "Walk", 'notes': "1 km = 1+ (calculated automatically)", 'up': True, 'down': False, 'priority': 2}
-  walk_habit_id = create_habit(user_id, walk_habit)
+  bike_walk_habit = {'type': "habit", 'text': "Bike and Walk", 'notes': "Automatically get points for every 1 km walked or 3 km biked. ***=== DO NOT EDIT BELOW THIS POINT ===*** AUTOCHECK: {\"mapper\": \"invite_friends\", \"args\": {\"walk_scale\": 1000, \"bike_scale\": 3000}}", 'up': True, 'down': False, 'priority': 2}
+  bike_walk_habit_id = create_habit(user_id, bike_walk_habit)
+  invite_friends = {'type': "habit", 'text': "Spread the word", 'notes': "Get points for inviting your friends! We're better together. ***=== DO NOT EDIT BELOW THIS POINT ===*** AUTOCHECK: {\"mapper\": \"invite_friends\", \"args\": {}}", 'up': True, 'down': False, 'priority': 2}
```

Comment:
before we merge this, you need to implement the `invite_friends` mapper, otherwise the executor will crash. I am not sure that the executor is the right way to implement this challenge since we don't store the invitations in the timeseries.

The executor only really works to map from the timeseries -> points. So we either store join group events in the timeseries, or we just modify the state directly from the proxy (in which case we cannot safely reset the user and restore their points).

Note also that if people join groups directly through habitica, the referral URL won't track anything...